plugins/org.locationtech.udig.core/src/org/locationtech/udig/core/internal/FeatureUtils.java

```diff
@@ -638,6 +641,53 @@ private static boolean toLinearRing( Geometry geom, Class< ? extends Geometry> t
         return true;
     }
 
+	/**
+	 * Performs a case insensitive lookup for the provided propertyName in the {@link FeatureType}.
+	 * 
+	 * @param featureType
+	 * @param propertyName
+	 * @return the actual propertyName (case sensitive) or null if none is found  
+	 */
+	public static String getActualPropertyName(final SimpleFeatureType featureType, final String propertyName) {
+
+		assert featureType != null : "featureType cannot be null";
+		assert propertyName != null : "propertyName != cannot be null";
+
+		for (int i = 0; i < featureType.getAttributeCount(); i++) {
+			String name = featureType.getDescriptor(i).getLocalName();
+
+			if (propertyName.equalsIgnoreCase(name)) {
+				return name;
+			}
+		}
+		return null;
+	}
+
+	
+	/**
+	 * Performs a case insensitive lookup for the provided propertyNames in the {@link FeatureType}.
+	 * 
+	 * @param featureType
+	 * @param propertyName
+	 * @return a list of actual propertyName (case sensitive). If a propertyName is not present
+	 * 			then it is skipped from the list  
+	 */
+	public static List<String> getActualPropertyName(final SimpleFeatureType featureType, final List<String> propertyNames) {
```

Comment:
No, it is a util function for possible future usage. after all this class contains util feature functions